userguide/storage.rst

```diff
@@ -713,53 +713,70 @@ existing datasets to close them--the remaining
-storage. **In most cases, compression provides storage gains
-comparable to deduplication with less impact on performance.**
-
-In %brand%, deduplication can be enabled during dataset creation. Be
-forewarned that **there is no way to undedup the data within a dataset
-once deduplication is enabled**, as disabling deduplication has
-**NO EFFECT** on existing data. The more data written to a
-deduplicated dataset, the more RAM it requires. When the system starts
-storing the DDTs (dedup tables) on disk because they no longer fit
-into RAM, performance craters. Further, importing an unclean pool can
-require between 3-5 GB of RAM per terabyte of deduped data, and if the
-system does not have the needed RAM, it will panic. The only solution
-is to add more RAM or recreate the pool.
-**Think carefully before enabling dedup!**
-This `article
-<http://constantin.glez.de/blog/2011/07/zfs-dedupe-or-not-dedupe>`_
-provides a good description of the value versus cost considerations
-for deduplication.
-
-**Unless a lot of RAM and a lot of duplicate data is available, do not
-change the default deduplication setting of "Off".**
-For performance reasons, consider using compression rather than
-turning this option on.
-
-If deduplication is changed to *On*, duplicate data blocks are removed
-synchronously. The result is that only unique data is stored and
-common components are shared among files. If deduplication is changed
-to *Verify*, ZFS will do a byte-to-byte comparison when two blocks
-have the same signature to make sure that the block contents are
-identical. Since hash collisions are extremely rare, *Verify* is
-usually not worth the performance hit.
-
-.. note:: After deduplication is enabled, the only way to disable it
-   is to use the :samp:`zfs set dedup=off {dataset_name}` command
-   from :ref:`Shell`. However, any data that has already been
-   deduplicated will not be un-deduplicated. Only newly stored data
-   after the property change will not be deduplicated. The only way to
-   remove existing deduplicated data is to copy all of the data off of
-   the dataset, set the property to off, then copy the data back in
-   again. Alternately, create a new dataset with
-   :guilabel:`ZFS Deduplication` left disabled, copy the data to the
-   new dataset, and destroy the original dataset.
+#ifdef freenas
+.. index:: Deduplication
+.. _Deduplication:
+
+Deduplication
+^^^^^^^^^^^^^
+
+Deduplication ("dedup") is a process where ZFS aims to store a single copy
+of any data that is repeated in different blocks or files, rather than storing 
+identical copies each time the data occurs. Deduplication 
+potentially allows repeated and identical data to be stored very 
+efficiently in a smaller space, but also places a 
+very heavy burden on RAM and on the CPU. In particular, dedup 
+significantly reduces write performance and usually requires large (and at 
+times huge) amounts of RAM.  See :ref:`Deduplication RAM requirements` for details.
```

Comment:
Get rid of usually and the aside in the parens. State the amount of RAM required, which is then followed up by the link to the RAM requirements.

Reply:
1) done

2) You can't "state the amount of RAM required - it depends totally on the extent of duplicate data and the structure/size of the pool. Some systems will need 5GB per TB, some will need 1GB per TB. Mine currently has a "dedup" and "nondedup" dataset in the pool, to bypass dedup processing for data I know won't be likely to deduplicate - the overall pool requirement is about 1.6GB per TB. You also need to consider how much RAM will be reserved for ARC and within that how much ARC is reserved for metadata - a user with 32GB RAM considering dedup might quite reasonably buy another 32GB and dedicate it 100% to metadata, while a different user might keep the 25% default ratio. 

**TL;DR you can't do what you're asking in the 2nd sentence.** All you can say here, is that dedup needs a lot (and in some cases a huge amount) of RAM and see below for more exact details how to work out the requirement for *your* data, and what you can do to trim it.

Reply:
1) done

2) You can't "state the amount of RAM required" - it totally depends on the extent of duplicate data and the structure/size of the pool. Some systems will need 5GB or more per TB, some will need <1GB per TB. Mine currently has a "dedup" and "nondedup" dataset in the pool, to bypass dedup processing for data I know won't be likely to deduplicate - the overall pool requirement is about 1.6GB per TB. You also need to consider how much RAM will be reserved for ARC and within that how much ARC is reserved for metadata - a user with 32GB RAM considering dedup might quite reasonably buy another 32GB and dedicate it 100% to metadata, while a different user might keep the 25% default ratio. 

**TL;DR you can't do what you're asking in the 2nd sentence.** All you can say here, is that dedup needs a lot (and in some cases a huge amount) of RAM and see below for more exact details how to work out the requirement for *your* data, and what you can do to trim it.